src/cargo/core/compiler/job_queue.rs

```diff
@@ -651,25 +635,33 @@ impl<'cfg> DrainState<'cfg> {
         // successful and otherwise wait for pending work to finish if it failed
         // and then immediately return.
         let mut error = None;
+        // CAUTION! From here on out, do not use `?`. Every error must be handled
```

Comment:
Could this loop get factored into a function that doesn't return `Result` to statically prevent usage of `?`?

Reply:
Updated, WDYT?  I assume `Result` will never be changed to implement `Try` such that it calls `result.ok()` to convert a Result to Option automatically?